04-onion-routing.md

```diff
@@ -511,15 +511,15 @@ If the onion version byte is unknown:
 2. data:
    * [32:sha256-of-onion]
 
-If the ephemeral key in the onion is unparsable:
+If the onion HMAC is incorrect:
```

Comment:
I don't think HMAC is the first check, if we can't parse the ephemeral key, then we can't generate the secret needed for the HMAC verification. So the check of the ephemeral key is implicitly done first. But let's not bike-shed this, I'm fine with either order :wink: